CMakeLists.txt

```diff
@@ -17,6 +17,11 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 project(VCU-GUI VERSION 1.1.0)
 set(TARGET_NAME ${PROJECT_NAME})
 
+# Must define '_USE_MATH_DEFINES' before #include <cmath>.
+# See https://learn.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170
+# We define it globally here.
+add_compile_definitions(_USE_MATH_DEFINES=1)
```

Comment:
Could you move this to `target_compile_definitions` with the other global defines?

See for example: https://dev.to/iblancasa/learning-cmake-2-43ga

Reply:
I've done this please double check my commit message I think I've followed the convention document but let me know if I haven't

Reply:
also I'm not sure about the checklist as I don't know how to use trunk so will probably need to sit down with you and see

Reply:
> I've done this please double check my commit message I think I've followed the convention document but let me know if I haven't

Re commit message, that's perfect. The convention is pretty simple, the main thing is to define the "type" and use common sense to set the "scope". There's lots of information in the actual conventional commits specification.

Reply:
> also I'm not sure about the checklist as I don't know how to use trunk so will probably need to sit down with you and see

Trunk will only run on WSL (at the moment), which is very unfortunate.
Unfortunately there is no equivalent "meta-linter" for Windows that I know of.

Possible solutions:
- Use WSL for all git operations on the repo and run trunk from there.
- Install all tools managed by Trunk on Windows (`clang-tidy` and `clang-format`), and run them manually without the benefits of Trunk.
- Set up CI/CD on GitHub to automatically do these checks with the [Trunk action for GitHub](https://github.com/trunk-io/trunk-action).